operate/webapp/src/main/java/io/camunda/operate/webapp/security/BaseWebConfigurer.java

```diff
@@ -75,14 +76,29 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
     return http.build();
   }
 
-  protected void applySecurityHeadersSettings(HttpSecurity http) throws Exception {
+  protected void applySecurityHeadersSettings(final HttpSecurity http) throws Exception {
     final WebSecurityProperties webSecurityConfig = operateProperties.getWebSecurity();
-    http.headers()
-        .contentSecurityPolicy(webSecurityConfig.getContentSecurityPolicy())
-        .and()
-        .httpStrictTransportSecurity()
-        .maxAgeInSeconds(webSecurityConfig.getHttpStrictTransportSecurityMaxAgeInSeconds())
-        .includeSubDomains(webSecurityConfig.getHttpStrictTransportSecurityIncludeSubDomains());
+
+    // Only SaaS has CloudProperties
+    if (operateProperties.getCloud().getClusterId() == null) {
+      webSecurityConfig.setContentSecurityPolicy(DEFAULT_SM_SECURITY_POLICY);
```

Comment:
I would prefer to handle the `webSecurityConfig` as an unmutable object because other clients of this object don't expect a change in a configuration object. WDYT?

Reply:
I agree